### PR TITLE
Add context to the Copilot Chat

### PR DESCRIPTION
**Pull request by lukka**
@benmcmorran @spebl @hnrqbaggio FYI

### CODE REVIEW

File: Extension/src/LanguageServer/client.ts
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/client.ts]: Is there a reason for the result variable versus just returning it?

**Review by sean-mcmanus — DISMISSED** [review]

File: Extension/src/LanguageServer/extension.ts
Comment (benmcmorran) [inline Extension/src/LanguageServer/extension.ts:259]: Will we have a way to measure usage of `#cpp`? I'm wondering if we should add our own telemetry, or if VS Code will emit some automatically.

File: Extension/package.json
Comment (bobbrow) [inline Extension/package.json:43]: We will need approval to use these APIs before this PR can be approved.

**Review by bobbrow — CHANGES_REQUESTED** [review]
(not actually requesting changes, just blocking the PR until we have approval for the API proposals)

File: Extension/src/LanguageServer/client.ts
Comment (lukka) [reply Extension/src/LanguageServer/client.ts]: removed the local var.

File: Extension/src/LanguageServer/extension.ts
Comment (lukka) [reply Extension/src/LanguageServer/extension.ts:259]: usage telemetry added

File: Extension/src/LanguageServer/extension.ts
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/extension.ts]: There's an extra space here -- not sure if that matters though: `${targetPlatform} .`

Comment (lukka) [reply Extension/src/LanguageServer/extension.ts]: fixed, thank you!

File: Extension/src/LanguageServer/extension.ts
Comment (jogo-) [inline Extension/src/LanguageServer/extension.ts:260]: duplicated word "the the"

**Comment by sean-mcmanus** [comment]
@lukka Did you want to get these changes into our next release 1.20.2?

**Comment by lukka** [comment]
> @lukka Did you want to get these changes into our next release 1.20.2?

Not yet, thank you for the heads up!

**Comment by lukka** [comment]
closing in favor of #12577

**Pull request closed without merging**
